### ISSUE TITLE
Missing parentheses at FunctionExpression inside CallExpression inside SequenceExpression

### ISSUE THREAD

**Issue by leumasme** [benjamn/recast#977]
The following Code will generate a Function Expression inside of a Call Expression inside of a Sequence Expression.
```(function(){})(), stuff()```
When parsed and prettyPrinted through recast, it will print out the following, syntax-invalid, code:
```function() {}(), stuff();```

**Comment by eventualbuddha** [comment]
Yep, reproduced this as follows:
```ts
const recast = require('recast')
const ast = recast.parse('(function(){})(), stuff()')
recast.parse(recast.prettyPrint(ast).code) // throws 'Unexpected token ('
```

**Comment by gnprice** [comment]
This issue no longer reproduces :tada:.

It was fixed by #1057: the issue still reproduces at the commit before that, and no longer does at that commit.
